lib/internal/test_runner/mock/mock_timers.js

```diff
@@ -98,7 +101,75 @@ class MockTimers {
     this.#executionQueue.removeAt(position);
   }
 
-  async * #setIntervalPromisified(interval, startTime, options) {
+  #createDate() {
+    kMock = Symbol('MockTimers');
+    /**
+     * Function to mock the Date constructor, treats cases as per ECMA-262
+     * and returns a Date object with a mocked implementation
+     * @typedef {Date} MockDate
+     * @returns {MockDate} a mocked Date object
+     */
+    function MockDate(year, month, date, hours, minutes, seconds, ms) {
+      const mockTimersSource = MockDate[kMock];
+      const nativeDate = mockTimersSource.#realDate;
+      // As of the fake-timers implementation for Sinon
+      // ref https://github.com/sinonjs/fake-timers/blob/a4c757f80840829e45e0852ea1b17d87a998388e/src/fake-timers-src.js#L456
+      // This covers the Date constructor called as a function ref.
+      // ECMA-262 Edition 5.1 section 15.9.2.
+      // and ECMA-262 Edition 14 Section 21.4.2.1
+      if (!(this instanceof MockDate)) {
+        return new nativeDate(mockTimersSource.#now).toString();
+      }
+
+
+
+      // Cases where Date is called as a constructor
+      switch (arguments.length) {
```

Comment:
shouldn't it be replaced by: 

```
const args = Array.from(arguments);
return new nativeDate(...args);
```

Reply:
I thought it too, however as per the [source](https://github.com/sinonjs/fake-timers/blob/a4c757f80840829e45e0852ea1b17d87a998388e/src/fake-timers-src.js#L456) in fake timers, which was the original inspiration. This is intended as mostly a defensive approach to avoid extra arguments and keeping the returns predictable.

Reply:
In fake-timers we didn't have `...` in the language when the code was written :)

Though the no argument defaults to current epoch case would still need to be explicit